wandb/sdk/data_types/saved_model.py

```diff
@@ -0,0 +1,329 @@
+import os
+from typing import (
+    Any,
+    cast,
+    ClassVar,
+    Generic,
+    Optional,
+    Type,
+    TYPE_CHECKING,
+    TypeVar,
+    Union,
+)
+
+
+import wandb
+from wandb import util
+from wandb.sdk.interface.artifacts import b64_string_to_hex, md5_files_b64
+
+from ._private import MEDIA_TMP
+from .base_types.wb_value import WBValue
+
+if TYPE_CHECKING:  # pragma: no cover
+    from wandb.apis.public import Artifact as PublicArtifact
+
+    from ..wandb_artifacts import Artifact as LocalArtifact
+    from ..wandb_run import Run as LocalRun
+
+    import cloudpickle  # type: ignore
+    import torch  # type: ignore
+    import sklearn  # type: ignore
+    import tensorflow  # type: ignore
+
+
+DEBUG_MODE = False
+
+
+SavedModelObjType = TypeVar("SavedModelObjType")
+
+
+class SavedModel(WBValue, Generic[SavedModelObjType]):
+    """SavedModel is a data type that can be used to store a model object
+    inside of a W&B Artifact.
+
+    _model_type_id: (str) The id of the SavedModel subclass used to serialize the model.
+    """
+
+    _log_type: ClassVar[str]
+    _path_extension: ClassVar[str]
+
+    _model_obj: Optional["SavedModelObjType"]
+    _path: Optional[str]
+    _input_obj_or_path: Union[SavedModelObjType, str]
+
+    # Public Methods
+    def __init__(self, obj_or_path: Union[SavedModelObjType, str]) -> None:
+        super(SavedModel, self).__init__()
+        if self.__class__ == SavedModel:
+            raise TypeError(
+                "Cannot instantiate abstract SavedModel class - please use SavedModel.init(...) instead."
+            )
+        self._model_obj = None
+        self._path = None
+        self._input_obj_or_path = obj_or_path
+        input_is_path = isinstance(obj_or_path, str) and os.path.exists(obj_or_path)
+        if input_is_path:
+            assert isinstance(obj_or_path, str)  # mypy
+            self._set_obj(self._deserialize(obj_or_path))
+        else:
+            self._set_obj(obj_or_path)
+
+        self._copy_to_disk()
```

Comment:
Can we add a comment after `_copy_to_disk()` that `self._path` should now be set to a temp path on the user's filesystem that contains all of their serialized model files?